src/erc20/filters/labeler.ts

```diff
@@ -0,0 +1,40 @@
+import { LabelerFilter, IEventLabel } from '../../interfaces';
+import { IEventData, EventKind } from '../types';
+
+function fmtAddr(addr: string) {
+  if (!addr) return '';
+  if (addr.length < 16) return addr;
+  return `${addr.slice(0, 7)}…${addr.slice(addr.length - 3)}`;
+}
+
+/**
+ * This a labeler function, which takes event data and describes it in "plain english",
+ * such that we can display a notification regarding its contents.
+ */
+export const Label: LabelerFilter = (
+  blockNumber: number,
+  chainId: string,
+  data: IEventData
+): IEventLabel => {
+  switch (data.kind) {
+    case EventKind.Approval:
+      return {
+        heading: 'Approval',
+        label: `Owner ${fmtAddr(data.owner)} approved spender ${fmtAddr(
+          data.spender
+        )}
+        to spend ${data.value}`,
+      };
+    case EventKind.Transfer:
+      return {
+        heading: 'Transfer',
+        label: `Transfer of ${data.value} from ${data.from} to ${data.to}.`,
```

Comment:
We may also want to figure out how to format these values in terms of the token symbol, if possible. Likely via optional arg that we pass through.